gobblin-data-management/src/main/java/gobblin/data/management/retention/profile/ConfigurableGlobDataset.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.retention.profile;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+
+import azkaban.utils.Props;
+
+import gobblin.data.management.retention.DatasetCleaner;
+import gobblin.data.management.retention.dataset.ConfigurableDataset;
+import gobblin.data.management.retention.dataset.Dataset;
+import gobblin.data.management.retention.dataset.finder.DatasetFinder;
+import gobblin.data.management.retention.version.DatasetVersion;
+
+
+/**
+ * A configurable {@link gobblin.data.management.retention.dataset.finder.DatasetFinder} that looks for
+ * {@link gobblin.data.management.retention.dataset.Dataset}s using a glob pattern.
+ */
+public class ConfigurableGlobDataset implements DatasetFinder {
```

Comment:
I think the class needs to be named ConfigurableGlobDatasetFinder since it implements the DatasetFinder interface, not the Dataset interface.